File: android/src/main/java/name/zeno/android/util/PrintUtils.java
package name.zeno.android.util;

import android.Manifest;
import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;
import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.RequiresPermission;
import android.text.TextUtils;

import java.util.Set;

import name.zeno.android.app.AppInfo;

/**
 * 打印工具类
 * Create Date: 16/6/9
 *
 * @author 陈治谋 (513500085@qq.com)
 */
public class PrintUtils
{
  private static final String FILENAME                         = "bt";
  private static final String DEFAULT_BLUETOOTH_DEVICE_ADDRESS = "default_bluetooth_device_address";
  private static final String DEFAULT_BLUETOOTH_DEVICE_NAME    = "default_bluetooth_device_name";

  public static final String ACTION_PRINT_TEST     = "action_print_test";
  public static final String ACTION_PRINT          = "action_print";
  public static final String ACTION_PRINT_TICKET   = "action_print_ticket";
  public static final String ACTION_PRINT_BITMAP   = "action_print_bitmap";
  public static final String ACTION_PRINT_PAINTING = "action_print_painting";

  public static final String PRINT_EXTRA = "print_extra";

  /**
   * 设置默认蓝牙设备地址
   */
  public static void setDefaultBluetoothDeviceAddress(Context mContext, String value)
  {
    SharedPreferences sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
    SharedPreferences.Editor editor = sharedPreferences.edit();
    editor.putString(DEFAULT_BLUETOOTH_DEVICE_ADDRESS, value);
    editor.apply();
    AppInfo.btAddress = value;
  }

  public static void setDefaultBluetoothDeviceName(Context mContext, String value)
  {
    SharedPreferences sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
    SharedPreferences.Editor editor = sharedPreferences.edit();
    editor.putString(DEFAULT_BLUETOOTH_DEVICE_NAME, value);
    editor.apply();
    AppInfo.btName = value;
  }

  @RequiresPermission(Manifest.permission.BLUETOOTH)
  public static boolean isBondPrinter(Context mContext, BluetoothAdapter bluetoothAdapter)
  {
    if (!BluetoothUtils.isOpen(bluetoothAdapter)) {
      return false;
    }
    String defaultBluetoothDeviceAddress = getDefaultBluetoothDeviceAddress(mContext);
    if (TextUtils.isEmpty(defaultBluetoothDeviceAddress)) {
      return false;
    }
    Set<BluetoothDevice> deviceSet = bluetoothAdapter.getBondedDevices();
    if (deviceSet == null || deviceSet.isEmpty()) {
      return false;
    }
    for (BluetoothDevice bluetoothDevice : deviceSet) {
      if (bluetoothDevice.getAddress().equals(defaultBluetoothDeviceAddress)) {
        return true;
      }
    }
    return false;

  }

  public static String getDefaultBluetoothDeviceAddress(Context mContext)
  {
    SharedPreferences sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
    return sharedPreferences.getString(DEFAULT_BLUETOOTH_DEVICE_ADDRESS, "");
  }

  public static boolean isBondPrinterIgnoreBluetooth(Context mContext)
  {
    String defaultBluetoothDeviceAddress = getDefaultBluetoothDeviceAddress(mContext);
    return !(TextUtils.isEmpty(defaultBluetoothDeviceAddress)
        || TextUtils.isEmpty(getDefaultBluetoothDeviceName(mContext)));
  }

  public static String getDefaultBluetoothDeviceName(Context mContext)
  {
    SharedPreferences sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
    return sharedPreferences.getString(DEFAULT_BLUETOOTH_DEVICE_NAME, "");
  }

}

File: library/src/main/java/name/zeno/android/util/PrintUtils.kt
package name.zeno.android.util

import android.Manifest
import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.content.Context
import android.content.SharedPreferences
import android.support.annotation.RequiresPermission
import android.text.TextUtils

import name.zeno.android.app.AppInfo

/**
 * 打印工具类
 * Create Date: 16/6/9
 *
 * @author 陈治谋 (513500085@qq.com)
 */
object PrintUtils {
  private val FILENAME = "bt"
  private val DEFAULT_BLUETOOTH_DEVICE_ADDRESS = "default_bluetooth_device_address"
  private val DEFAULT_BLUETOOTH_DEVICE_NAME = "default_bluetooth_device_name"

  val ACTION_PRINT_TEST = "action_print_test"
  val ACTION_PRINT = "action_print"
  val ACTION_PRINT_TICKET = "action_print_ticket"
  val ACTION_PRINT_BITMAP = "action_print_bitmap"
  val ACTION_PRINT_PAINTING = "action_print_painting"

  val PRINT_EXTRA = "print_extra"

  /**
   * 设置默认蓝牙设备地址
   */
  fun setDefaultBluetoothDeviceAddress(mContext: Context, value: String) {
    val sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
    val editor = sharedPreferences.edit()
    editor.putString(DEFAULT_BLUETOOTH_DEVICE_ADDRESS, value)
    editor.apply()
    AppInfo.btAddress = value
  }

  fun setDefaultBluetoothDeviceName(mContext: Context, value: String) {
    val sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
    val editor = sharedPreferences.edit()
    editor.putString(DEFAULT_BLUETOOTH_DEVICE_NAME, value)
    editor.apply()
    AppInfo.btName = value
  }

  @RequiresPermission(Manifest.permission.BLUETOOTH)
  fun isBondPrinter(mContext: Context, bluetoothAdapter: BluetoothAdapter): Boolean {
    if (!BluetoothUtils.isOpen(bluetoothAdapter)) {
      return false
    }
    val defaultBluetoothDeviceAddress = getDefaultBluetoothDeviceAddress(mContext)
    if (TextUtils.isEmpty(defaultBluetoothDeviceAddress)) {
      return false
    }
    val deviceSet = bluetoothAdapter.bondedDevices
    if (deviceSet == null || deviceSet.isEmpty()) {
      return false
    }
    for (bluetoothDevice in deviceSet) {
      if (bluetoothDevice.address == defaultBluetoothDeviceAddress) {
        return true
      }
    }
    return false

  }

  fun getDefaultBluetoothDeviceAddress(mContext: Context): String {
    val sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
    return sharedPreferences.getString(DEFAULT_BLUETOOTH_DEVICE_ADDRESS, "")
  }

  fun isBondPrinterIgnoreBluetooth(mContext: Context): Boolean {
    val defaultBluetoothDeviceAddress = getDefaultBluetoothDeviceAddress(mContext)
    return !(TextUtils.isEmpty(defaultBluetoothDeviceAddress) || TextUtils.isEmpty(getDefaultBluetoothDeviceName(mContext)))
  }

  fun getDefaultBluetoothDeviceName(mContext: Context): String {
    val sharedPreferences = mContext.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
    return sharedPreferences.getString(DEFAULT_BLUETOOTH_DEVICE_NAME, "")
  }

}
